### ISSUE TITLE
Swap

### ISSUE THREAD

**Issue by jeremykentbgross** [dharmafly/jsonpatch.com#53]
This json patch standard is awesome.  However it should also have a swap operation also imo.

Example:

json:
{
&nbsp;&nbsp;&nbsp;&nbsp; myArray: [1, 2],
&nbsp;&nbsp;&nbsp;&nbsp; myObject:{ a:1, b: 2},
&nbsp;&nbsp;&nbsp;&nbsp; myString: "Testing"
}

patch:
[
&nbsp;&nbsp;&nbsp;&nbsp; { op: 'swap', from: 'myArray/0', path: 'myArray/1' }
&nbsp;&nbsp;&nbsp;&nbsp; { op: 'swap', from: 'myObject/a', path: 'myObject/b' }
&nbsp;&nbsp;&nbsp;&nbsp; { op: 'swap', from: 'myArray/0', path: 'myString' }
]

new json:
{
&nbsp;&nbsp;&nbsp;&nbsp; myArray: ["Testing", 1],
&nbsp;&nbsp;&nbsp;&nbsp; myObject:{ a:2, b: 1},
&nbsp;&nbsp;&nbsp;&nbsp; myString: 2
}

**Comment by almost** [comment]
I see how that could be useful, you can sort of do it with three moves buy it requires a "temporary" location to keep the value in.

```
patch:
[
     { op: 'move', from: 'myArray/0', path: 'temp-path-name' }
     { op: 'move', from: 'myArray/1', path: '/myArray/0' }
     { op: 'move', from: 'temp-path-name', path: '/myArray/1' }
]
```

The RFC for json-patch is here: https://tools.ietf.org/html/rfc6902 so you'd need to go through he IETF procedures to get that added. More likely they already discussed something similar and decided not to include it. There are probably archives of the mailing list somewhere but I'm not sure where.

**Comment by jeremykentbgross** [comment]
Yeah I already thought of the temp path method.  However I think in my tool I will implement a swap op anyway.  I just prefer to follow the standards as much as possible and it seems to me that this is an obvious enough case others must want/need it so I figured I would mention.  I didn't realize this was from another external standard.

Looks like move actually does effectively a splice in the case of an array instead of overwriting the original.  Guess it's a good thing I followed your link because I didn't realize that.
